odoo/addons/base/models/image_mixin.py

```diff
@@ -0,0 +1,109 @@
+
+    @api.multi
+    @api.depends('image_raw_original')
+    def _compute_image_raw(self):
+        for record in self:
+            image_raw_original = record.image_raw_original
+            record.image_raw_original_byte_size = len(image_raw_original) if image_raw_original else 0
+            if isinstance(image_raw_original, str):
+                image_raw_original = image_raw_original.encode('ascii')
+            record.image_raw_big = tools.image_resize_image_big(image_raw_original, avoid_if_small=True)
+            record.image_raw_large = tools.image_resize_image_large(image_raw_original, avoid_if_small=True)
+            record.image_raw_medium = tools.image_resize_image_medium(image_raw_original, avoid_if_small=True)
+            record.image_raw_small = tools.image_resize_image_small(image_raw_original, avoid_if_small=True)
+
+    @api.multi
+    def _compute_image_original(self):
+        for record in self:
+            record.image_original = record.image_raw_original or record._get_image_fallback_record().image_raw_original
+
+    @api.multi
+    def _set_image_original(self):
+        for record in self:
+            fallback = record._get_image_fallback_record()
+            if not record.image_original and not record.image_raw_original:
+                # If we are trying to remove an image even though it is already
+                # not set, remove it from the fallback record instead.
+                fallback.image_raw_original = False
+            elif record.image_original and not record._get_image_fallback_record().image_raw_original:
+                # If we are trying to add an image, but the fallback image is
+                # not set, write on the fallback instead.
+                fallback.image_raw_original = record.image_original
+            else:
+                record.image_raw_original = record.image_original
+
+    @api.multi
+    def _compute_image_big(self):
+        for record in self:
+            record.image_big = record.image_raw_big or record._get_image_fallback_record().image_raw_big
+
+    @api.multi
+    def _compute_image_large(self):
+        for record in self:
+            record.image_large = record.image_raw_large or record._get_image_fallback_record().image_raw_large
+
+    @api.multi
+    def _compute_image_medium(self):
+        for record in self:
+            record.image_medium = record.image_raw_medium or record._get_image_fallback_record().image_raw_medium
+
+    @api.multi
+    def _compute_image_small(self):
+        for record in self:
+            record.image_small = record.image_raw_small or record._get_image_fallback_record().image_raw_small
+
+    @api.multi
+    @api.depends('image_original')
+    def _compute_image_original_byte_size(self):
+        for record in self:
+            record.image_original_byte_size = len(record.image_original) if record.image_original else 0
```

Comment:
use bin_size and pg_size_pretty

Reply:
I don't want the size pretty here, this is just a value that I need to compute if the badge should be green, yellow, or red. It's easier to compare int than to parse a string.

The size is already prettied automatically when doing image.value (instead of raw_value) in the kanban view.

Reply:
btw since stored in filestore and no in db, psql will not help you ;)

Reply:
I've decided to remove the field... Since we already have the blob in JS, might as well just do `.length` on it.